exp-player/addon/mixins/exp-player-route.js

```diff
@@ -0,0 +1,43 @@
+import Ember from 'ember';
+
+export default Ember.Mixin.create({
+    _experiment: null,
+    _session: null,
+
+    currentUser: Ember.inject.service(),
+
+    _getExperiment(params) { // jshint ignore: line
+
+    },
+    _getSession(params, experiment) { // jshint ignore: line
+
+    },
+    model(params) {
+        // While a little gross, this ensures all the criteria for participation
+        // are met before the route resolves. This has the benefit that the route's
+        // loading state is active until all of this is complete.
+        return this._getExperiment(params).then((experiment) => {
```

Comment:
I definitely considered using a hash. There is some difficulty/complexity here with the dependent flow of data here (session needs experiment, pastSessions needs experiment + account + profile, etc).